onboard-server/src/main/java/no/vegvesen/ixn/serviceprovider/OnboardRestController.java

```diff
@@ -319,20 +331,36 @@ public RedirectView deletePrivateChannel(@PathVariable String serviceProviderNam
 	}
 
 	@RequestMapping(method = RequestMethod.GET, path = "/{serviceProviderName}/privatechannels", produces = MediaType.APPLICATION_JSON_VALUE)
-	public PrivateChannelListApi getPrivateChannels(@PathVariable String serviceProviderName) {
+	public ListPrivateChannelsResponse getPrivateChannels(@PathVariable String serviceProviderName) {
 		OnboardMDCUtil.setLogVariables(nodeProperties.getName(), serviceProviderName);
 		logger.info("listing private channels for service provider {}", serviceProviderName);
 		this.certService.checkIfCommonNameMatchesNameInApiObject(serviceProviderName);
 
-		ServiceProvider serviceProvider = getOrCreateServiceProvider(serviceProviderName);
-
-		List<PrivateChannel> privateChannels = serviceProvider.getPrivateChannels().stream().collect(Collectors.toList());
+		List<PrivateChannel> privateChannels = privateChannelRepository.findAllByServiceProviderName(serviceProviderName);
 		List<PrivateChannelApi> privateChannelsApis = new ArrayList<>();
-		for(PrivateChannel privateChannel : privateChannels){
+		for (PrivateChannel privateChannel : privateChannels) {
```

Comment:
Create new api response object for a single private channel, the queueName is not set yet and is returned as null in the addResponse. Also remember to set the name in the api response object